.github/workflows/tests.yml

```diff
@@ -0,0 +1,79 @@
+name: Tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  test_linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
```

Comment:
what's the opposite of "fail-fast"? (and what's the default?)

Reply:
if the default is true, should be omit this line? having a line that sets the value to the default is generally misleading (people will assume it's doing something intentional)

Reply:
Sounds good, I'll send a fix!